oap-server/server-storage-plugin/storage-banyandb-plugin/src/main/java/org/apache/skywalking/oap/server/storage/plugin/banyandb/MetadataRegistry.java

```diff
@@ -258,7 +229,7 @@ private List<TagMetadata> parseExtraTagSpecs(String tags, Schema.SchemaBuilder b
     private TagFamilySpec.TagSpec parseTagSpec(ModelColumn modelColumn) {
         final Class<?> clazz = modelColumn.getType();
         final String colName = modelColumn.getColumnName().getStorageName();
-        if (String.class.equals(clazz) || DataTable.class.equals(clazz) || JsonObject.class.equals(clazz)) {
+        if (String.class.equals(clazz) || StorageDataComplexObject.class.isAssignableFrom(clazz) || JsonObject.class.equals(clazz)) {
             return TagFamilySpec.TagSpec.newStringTag(colName);
```

Comment:
If this is a `StorageDataComplexObject` and `JsonObject`, it should not be indexed as a tag. A field is enough.
The String should follow `storageOnly` annotation attribute.

Reply:
We never search on this type of fields as it is meaningless.

Reply:
No. It does not work like this. An index rule determines whether a tag is indexed.

Reply:
Then, how should we determine a field or a no-index tag?

Reply:
> Then, how should we determine a field or a no-index tag?

Currently, every entity only have one single field. It is determined by `ValueColumnMetadata.getValueCName`

Reply:
So, the field is only for `groupby`'s aggregation function? @hanahmily What is the recommendation from the server-side?
I am feeling we need a doc definition about this, in the cases for further PR review like this, especially for new contributors.

Reply:
> So, the field is only for `groupby`'s aggregation function? @hanahmily What is the recommendation from the server-side? I am feeling we need a doc definition about this, in the cases for further PR review like this, especially for new contributors.

No. It can also be read, for example in `multiGet`

Reply:
Yes, it can be read, but as well as tag. The development doc for SkyWalking is too early, we even don't have a conclusion yet. Jaeger is easy, it just needs a span query.
It is better we put recommendation from server's perspective, then we determine which is better for now. We are getting to know more about the progress of this integration. We are very early.